packages/babel-traverse/src/path/family.js

```diff
@@ -18,6 +18,61 @@ function addCompletionRecords(path, paths) {
   return paths;
 }
 
+function completionRecordForSwitch(cases, paths) {
+  let isLastCaseWithConsequent = true;
+
+  for (let i = cases.length - 1; i >= 0; i--) {
+    const switchCase = cases[i];
+    const consequent = switchCase.get("consequent");
+
+    let breakStatement;
+    findBreak: for (const statement of consequent) {
+      if (statement.isBlockStatement()) {
+        for (const statementInBlock of statement.get("body")) {
+          if (statementInBlock.isBreakStatement()) {
+            breakStatement = statementInBlock;
+            break findBreak;
+          }
+        }
+      } else if (statement.isBreakStatement()) {
+        breakStatement = statement;
+        break;
+      }
+    }
+
+    if (breakStatement) {
+      while (
+        breakStatement.key === 0 &&
+        breakStatement.parentPath.isBlockStatement()
+      ) {
+        breakStatement = breakStatement.parentPath;
+      }
+
+      const prevSibling = breakStatement.getPrevSibling();
+      if (
+        breakStatement.key > 0 &&
+        (prevSibling.isExpressionStatement() || prevSibling.isBlockStatement())
+      ) {
+        paths = addCompletionRecords(prevSibling, paths);
+        breakStatement.remove();
+      } else {
+        breakStatement.replaceWith(switchCase.scope.buildUndefinedNode());
```

Comment:
nit: I think it should be `breakStatement.scope.buildUndefinedNode()`.

Reply:
@JLHwung sure!

Reply:
After #10243 they will be logically equivalent: `buildUndefinedNode` does not depend on scope any more. It is by reviewing this PR that I find the room for improvements in `buildUndefinedNode`.